### PR TITLE
chore: CL apptesting helpers for SQS

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR moves CL test helpers into the `apptesting` package so that these heleprs can be reused in SQS.

This is partial transfer of https://github.com/osmosis-labs/osmosis/pull/6785 where the CL apptesting suite is already in-use

### CODE REVIEW

File: app/apptesting/concentrated_liquidity.go [hunk 1]
```diff
@@ -6,21 +6,790 @@ import (
...
+	swapFundCoins = sdk.NewCoins(sdk.NewCoin("eth", osmomath.NewInt(10000000000000)), sdk.NewCoin("usdc", osmomath.NewInt(1000000000000)))
+
+	roundingError = osmomath.OneInt()
+	EmptyCoins    = sdk.DecCoins(nil)
+
+	// Various sqrt estimates
+	Sqrt4994 = osmomath.MustNewDecFromStr("70.668238976219012614")
+
+	// swap out given in test cases
+	SwapOutGivenInCases = map[string]ConcentratedSwapTest{
```
Comment (AlpinYukseloglu) [inline app/apptesting/concentrated_liquidity.go:104]: Side-note: one downside of using the map structure is it's not possible to add lists of test cases together when we want to test more general behavior

I wonder if there's a solution to this that doesn't include doing a `name` field which looks a bit messy?

**Review by AlpinYukseloglu — APPROVED** [review]
Wow, looks like this was a pretty heavy test refactor. Might be worthwhile to look into whether we should be leaning into apptesting more for future features to avoid these kinds of large changes.

LGTM

**Review by mattverse — COMMENTED** [review]
ACK, nice! Must have took some time to do this refactor 030

**Comment by p0mvn** [comment]
@mergifyio refresh

**Pull request merged**